odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/IngestionServiceImpl.java

```diff
@@ -120,12 +120,9 @@ public Mono<Void> ingest(final DataEntityList dataEntityList) {
     }
 
     private Mono<Long> acquireDataSourceId(final String dataSourceOddrn) {
```

Comment:
it's just a suggestion, but you can rewrite all of this lambda logic into something

```
Mono<Long> injectedIdMono = Mono.just(dataSourceOddrn)
            .map(oddrnGenerator::parse)
            .filter(Optional::isPresent)
            .map(Optional::get)
            .onErrorMap(
                e -> new RuntimeException(String.format("Couldn't parse %s into OddrnPath", dataSourceOddrn), e))
            .filter(path -> path instanceof ODDPlatformDataSourcePath)
            .switchIfEmpty(
                Mono.error(() -> new NotFoundException("Data source with oddrn %s hasn't been found", dataSourceOddrn)))
            .map(path -> ((ODDPlatformDataSourcePath) path).getDatasourceId())
            .flatMap(id -> dataSourceRepository.injectOddrn(id, dataSourceOddrn))
            .map(dataSourceDto -> dataSourceDto.dataSource().getId());
```

and do switchIfEmpty(injectedIdMono)

I didn't test it, so it's up to you